File: pptx-workspace/screenshot.js
const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

const out = path.join(__dirname, 'screenshots');
fs.mkdirSync(out, { recursive: true });

(async () => {
    const browser = await chromium.launch({ headless: true });
    const page = await browser.newPage({ viewport: { width: 1366, height: 768 } });
    await page.goto('https://neexios.github.io/christophe-de-margerie/', { waitUntil: 'networkidle' });

    // Screenshot loading/intro
    await page.screenshot({ path: path.join(out, '01-intro.png') });
    console.log('01-intro done');

    // Wait for loading screen to finish, then click button
    await page.waitForTimeout(5000);
    try {
        await page.click('text=Ouvrir le dossier', { timeout: 8000 });
    } catch (e) {
        console.log('Could not find button, clicking body');
        await page.click('body');
    }

    // Wait for dossier + newspaper animations
    await page.waitForTimeout(8000);
    await page.screenshot({ path: path.join(out, '02-after-intro.png') });
    console.log('02-after-intro done');

    // Scroll through each scene
    const scenes = ['scene-1', 'scene-2', 'scene-3', 'scene-4', 'scene-5', 'scene-6', 'scene-7', 'scene-end'];
    for (let i = 0; i < scenes.length; i++) {
        await page.evaluate((sid) => {
            const el = document.getElementById(sid);
            if (el) el.scrollIntoView({ behavior: 'instant' });
        }, scenes[i]);
        await page.waitForTimeout(2000);
        const num = String(i + 3).padStart(2, '0');
        await page.screenshot({ path: path.join(out, `${num}-${scenes[i]}.png`) });
        console.log(`${num}-${scenes[i]} done`);
    }

    await browser.close();
    console.log('All screenshots saved!');
})();

File: pptx-workspace/screenshot.py
from playwright.sync_api import sync_playwright
import os

out = r"c:\Users\neoli\OneDrive\Desktop\Workspace-IA\Christophe De Margerie\pptx-workspace\screenshots"
os.makedirs(out, exist_ok=True)

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport={"width": 1366, "height": 768})
    page.goto("https://neexios.github.io/christophe-de-margerie/", wait_until="networkidle")

    # Screenshot the loading/intro screen
    page.screenshot(path=os.path.join(out, "01-intro.png"))

    # Click "Ouvrir le dossier" button
    page.wait_for_timeout(4000)  # wait for loading screen
    try:
        page.click("text=Ouvrir le dossier", timeout=10000)
    except:
        # Maybe loading screen still visible, click through it
        page.click("body")

    page.wait_for_timeout(6000)  # wait for dossier + newspapers animations

    # Screenshot scene 1
    page.screenshot(path=os.path.join(out, "02-scene1.png"))

    # Scroll through each scene and screenshot
    scenes = ["scene-1", "scene-2", "scene-3", "scene-4", "scene-5", "scene-6", "scene-7", "scene-end"]
    for i, sid in enumerate(scenes):
        page.evaluate(f'document.getElementById("{sid}")?.scrollIntoView({{behavior: "instant"}})')
        page.wait_for_timeout(1500)
        page.screenshot(path=os.path.join(out, f"{i+3:02d}-{sid}.png"))

    browser.close()
    print("Screenshots saved!")
